packages/stateful/actions/actions/AuthzAuthorization.tsx

```diff
@@ -0,0 +1,83 @@
+import type { MsgGrant, MsgRevoke } from 'cosmjs-types/cosmos/authz/v1beta1/tx'
+import { useCallback, useMemo } from 'react'
+
+import { KeyEmoji } from '@dao-dao/stateless'
+import {
+  ActionComponent,
+  ActionMaker,
+  CoreActionKey,
+  UseDecodedCosmosMsg,
+  UseDefaults,
+  UseTransformToCosmos,
+} from '@dao-dao/types/actions'
+import { makeStargateMessage } from '@dao-dao/utils'
+
+import { AuthzAuthorizationComponent as StatelessAuthzComponent } from '../components/AuthzAuthorization'
+
+interface AuthzData {
+  custom?: boolean
+  type_url: string
+  value: MsgGrant | MsgRevoke
+}
+
+const useDefaults: UseDefaults<AuthzData> = () => ({
+  custom: false,
+  type_url: '/cosmos.authz.v1beta1.MsgGrant',
+  value: {
+    grantee: '',
+    granter: '',
+    msgTypeUrl: '/cosmos.staking.v1beta1.MsgDelegate',
+  },
+})
+
+const Component: ActionComponent = (props) => {
+  return <StatelessAuthzComponent {...props} options={{}} />
+}
+
+const useDecodedCosmosMsg: UseDecodedCosmosMsg<AuthzData> = (
+  msg: Record<string, any>
+) =>
+  useMemo(
+    () =>
+      'stargate' in msg && msg.stargate.typeUrl && msg.stargate.value
+        ? {
+            match: true,
+            data: {
+              type_url: msg.stargate.type_url,
+              value: msg.stargate.value,
+            },
+          }
+        : { match: false },
+    [msg]
+  )
+
```

Comment:
the goal of this hook is to recognize from a CosmosMsg whether or not it is of this action type. right now, all stargate messages will try to render as authz authorization actions 🤪